lab1/BigInt.h

```
@@ -0,0 +1,780 @@
+
+        //division assignment 
+        BigInt & operator/=(const BigInt &);
+
+        //modulo assignment 
+        BigInt & operator%=(const BigInt &);
+
+        //bitwise XOR assignment 
+        BigInt & operator^=(const BigInt &);
+
+        //bitwise AND assignment 
+        BigInt & operator&=(const BigInt &);
+
+        //bitwise OR assignment
+        BigInt & operator|=(const BigInt &);
+};
+
+        BigInt::BigInt() {
+            this->isNegative_ = false;
+            this->digits_.push_back(0);
+        }
+
+        BigInt::BigInt(int num) {
+            if (num == 0) {
+                BigInt();
+            }
+            else {
+                if (num > 0) {
+                    this->isNegative_ = false;
+                }
+                if (num < 0) {
+                    this->isNegative_ = true;
+                    num = -num;
+                }
+                do {
+                    int reminder = num % BigInt::base_; 
+                    this->digits_.push_back(reminder);
+                    num /= BigInt::base_;
+                } while (num != 0);
+            }
+        }
+
+        void BigInt::check_string(std::string str) {
+            std::string tmp;
+            if (str[0] == '-') {
+                str = str.substr(1);
+            }
+            for (long long int i = 0; i < str.size(); i++) {
+                if (str[i] < '0' || str[i] > '9') {
+                    throw BigInt::invalid_argument();
+                    return;
+                }
+            }
+        }
+
+        void BigInt::remove_leading_zeros() {
+            while (this->digits_.size() > 1 && this->digits_.back() == 0) {
+                this->digits_.pop_back();
+            }
+            if (this->digits_.size() == 1 && this->digits_[0] == 0) {
+                this->isNegative_ = false;
+            }
+        }
+
+        BigInt::BigInt(std::string str) {
+            if (str.length() == 0) {
```

It's better to check this in check_string method (throwing invalid_argument in this case) instead of making double check logic